f5/bigip/mixins.py

```diff
@@ -316,3 +316,79 @@ def get_device_info(self, bigip):
         device = [device for device in coll if device.selfDevice == 'true']
         assert len(device) == 1
         return device[0]
+
+
+class StatsMixin(object):
+    """This mixin allows attaching 'stats' object as an
+        attribute to a resource.
+
+        Stats are available on some objects, and require
+        the target object to be loaded first.
+        It also allows the caller to retrieve the converted json
+        response in unprocessed state.
+
+    """
+
+    def _key_dot_replace(self, rdict):
+        """Replace fullstops in returned keynames"""
+        if isinstance(rdict, dict):
+            return {key.replace('.', '_'): self._key_dot_replace(val)
+                    for key, val in rdict.items()}
+
+    def _get_stats(self, rdict):
+        """Helper method to convert dictionary into an object
+
+            Each stat element in the returned json
+            contains a single key:value pair dictionary.
+            Hence this method will allow access to each stat value
+            using python 'dot' notation.
+        """
+
+        class DictConvert(object):
```

Comment:
Instead of creating a new object, you might consider inheriting from dict and overriding __getattr__. Then you'll get things like isinstance(x, dict) and __eq__ for free.